crates/swc_plugin_macro/src/lib.rs

```diff
@@ -116,11 +129,8 @@ fn handle_func(func: ItemFn) -> TokenStream {
                 return construct_error_ptr(err);
             }
 
-            (
-                0,
-                serialized_result.as_ptr() as _,
-                serialized_result_len.expect("Should be an i32"),
-            )
+            set_transform_result_volatile(serialized_result.as_ptr() as _, serialized_result_len.expect("Should be an i32"));
```

Comment:
this is main change in macro side. Instead of return value, it calls host imported fn first to set value. When it's being called, host's fn access env & allocate host side values.